eva/utils/single_dispatch_method.py

```diff
@@ -0,0 +1,19 @@
+import sys
+
+if sys.version_info >= (3, 8):
+    from functools import singledispatchmethod
+else:
+    # https://stackoverflow.com/questions/24601722/how-can-i-use-functools-singledispatch-with-instance-methods
+    from functools import singledispatch, update_wrapper
+
+    def singledispatchmethod(func):
```

Comment:
Add docstring describing what single dispatch is briefly and provide short code example.

Reply:
done.